front/src/common/InfraSelector/InfraSelectorEditionActionsBar.tsx

```diff
@@ -1,40 +1,51 @@
 import React, { useState } from 'react';
 import { FaCopy, FaDownload, FaLock, FaLockOpen, FaPencilAlt } from 'react-icons/fa';
 import { useTranslation } from 'react-i18next';
-import { post, get, put } from 'common/requests';
 import { MdCancel, MdCheck } from 'react-icons/md';
 import fileDownload from 'js-file-download';
 import { Infra, osrdEditoastApi } from 'common/api/osrdEditoastApi';
-import { INFRA_URL } from './Consts';
+import { useDispatch } from 'react-redux';
+import { setFailure } from 'reducers/main';
+import InfraLockState from './consts';
 
 type ActionBarProps = {
   infra: Infra;
   isFocused?: number;
   setIsFocused: (focus?: number) => void;
-  getInfrasList: () => void;
   inputValue: string;
 };
 
-export default function ActionsBar({
-  infra,
-  isFocused,
-  setIsFocused,
-  getInfrasList,
-  inputValue,
-}: ActionBarProps) {
+export default function ActionsBar({ infra, isFocused, setIsFocused, inputValue }: ActionBarProps) {
   const { t } = useTranslation('infraManagement');
   const [isWaiting, setIsWaiting] = useState(false);
+  const dispatch = useDispatch();
 
+  const [lockInfra] = osrdEditoastApi.usePostInfraByIdLockMutation();
+  const [unlockInfra] = osrdEditoastApi.usePostInfraByIdUnlockMutation();
+  const [getRailjson] = osrdEditoastApi.useLazyGetInfraByIdRailjsonQuery();
   const [cloneInfra] = osrdEditoastApi.usePostInfraByIdCloneMutation();
+  const [updateInfra] = osrdEditoastApi.usePutInfraByIdMutation();
 
-  async function handleLockedState(action: string) {
+  async function handleLockedState(action: InfraLockState) {
     if (!isWaiting) {
       setIsWaiting(true);
       try {
-        await post(`${INFRA_URL}${infra.id}/${action}/`, {});
-        getInfrasList();
-        setIsWaiting(false);
+        if (action === InfraLockState.LOCK) {
+          await lockInfra({ id: infra.id });
+        }
+        if (action === InfraLockState.UNLOCK) {
+          await unlockInfra({ id: infra.id });
+        }
       } catch (e) {
+        if (e instanceof Error) {
```

Comment:
We're sure that e will always be an instance of Error?

Reply:
Couldn't test it but it's coded like that in other components so I thought it was ok and still better to have a toast displayed to the user if an action fails.

Reply:
Ok I found something, let me know if you are ok